plugins/clipboard/msd-clipboard-manager.c

```diff
@@ -139,8 +139,9 @@ send_selection_notify (MsdClipboardManager *manager,
         notify.property = success ? manager->priv->property : None;
         notify.time = manager->priv->time;
 
-        display = gdk_display_get_default ();
-        gdk_x11_display_error_trap_push (display);
+        display = gdk_x11_lookup_xdisplay (manager->priv->display);
```

Comment:
This does not change anything. `priv->display` is `gdk_display_get_default()`:
https://github.com/mate-desktop/mate-settings-daemon/blob/96e15607e8081694b993b7f7d509ef7ab8c75b5a/plugins/clipboard/msd-clipboard-manager.c#L979-L986

Reply:
As said, most of it probably don't change anything as there probably is only one display anyway.  Yet, `gdk_display_get_default()`'s return value can theoretically change over time, so this *could* not be the same.  I don't think it ever really happens in most real situations, though.

Reply:
For that to happen someone would need to call `gdk_display_open` to create new display **and** close default display. Clipboard code is not doing that... Using `GtkInspector` you can get second display, but that won't change default display.

Reply:
Looking at the code, the simplest way would be calling [`gdk_display_manager_set_default_display()`](https://docs.gtk.org/gdk3/method.DisplayManager.set_default_display.html), but that's not something any more likely to happen.